app/src/main/java/org/fossasia/pslab/communication/sensors/Sx1276.java

```diff
@@ -0,0 +1,454 @@
+        }
+        SPIWrite(REG_FIFO, (ArrayList<Byte>) byteArray.subList(0, size));
+        setupBytesArray(currentLength + size);
+        SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+        return size;
+    }
+
+    public boolean available() throws IOException {
+        return (SPIRead(REG_RX_NB_BYTES, 1).get(0) - packetIndex) == 1;
+    }
+
+    public int checkRx() throws IOException {
+        byte irqFlags = SPIRead(REG_IRQ_FLAGS, 1).get(0);
+        if (((irqFlags & IRQ_RX_DONE_MASK) == 1) && ((irqFlags & IRQ_PAYLOAD_CRC_ERROR_MASK) == 1)) {
+            return 1;
+        } else {
+            return 0;
+        }
+    }
+
+    public byte read() throws IOException {
+        if (available()) {
+            packetIndex++;
+            return SPIRead(REG_FIFO, 1).get(0);
+        } else return -1;
+    }
+
+    public ArrayList<Byte> readAll() throws IOException {
+        ArrayList<Byte> p = new ArrayList<>();
+        while (available()) {
+            p.add(read());
+        }
+        return p;
+    }
+
+    public byte peek() throws IOException {
+        if (available()) {
+            ArrayList<Byte> currentAddress = SPIRead(REG_FIFO_ADDR_PTR, 1);
+            byte val = SPIRead(REG_FIFO, 1).get(0);
+            SPIWrite(REG_FIFO_ADDR_PTR, currentAddress);
+            return val;
+        } else {
+            return -1;
+        }
+    }
+
+    public void flush() {
+        /**/
+    }
+
+    public void receive(int size) throws IOException {
+        if (size > 0) {
+            implicitHeaderMode();
+            setupBytesArray(size & 0xFF);
+            SPIWrite(REG_PAYLOAD_LENGTH, bytes);
+        } else {
+            explicitHeaderMode();
+            setupBytesArray(MODE_LONG_RANGE_MODE | MODE_RX_SINGLE);
+            SPIWrite(REG_OP_MODE, bytes);
+        }
+    }
+
+    private void idle() throws IOException {
+        setupBytesArray(MODE_LONG_RANGE_MODE | MODE_STDBY);
+        SPIWrite(REG_OP_MODE, bytes);
+    }
+
+    private void sleep() throws IOException {
+        setupBytesArray(MODE_LONG_RANGE_MODE | MODE_SLEEP);
+        SPIWrite(REG_OP_MODE, bytes);
+    }
+
+    private void setTxPower(int level, int pin) throws IOException {
+        if (pin == PA_OUTPUT_RFO_PIN) {
+            if (level < 0) {
+                level = 0;
+            } else if (level > 14) {
+                level = 14;
+            }
+            setupBytesArray(0x70 | level);
+            SPIWrite(REG_PA_CONFIG, bytes);
+        } else {
+            if (level < 2) {
+                level = 2;
+            } else if (level > 17) {
+                level = 17;
+            }
+            if (level == 17) {
+                Log.d(name, "max power output");
+                setupBytesArray(0x87);
+                SPIWrite(REG_PA_DAC, bytes);
+            } else {
+                setupBytesArray(0x84);
+                SPIWrite(REG_PA_DAC, bytes);
+                setupBytesArray(PA_BOOST | 0x70 | (level - 2));
```

Comment:
As per the python repo, this line & the line below it are outside of this else statement.